website/source/docs/builders/virtualbox-vm.html.md.erb

```diff
@@ -0,0 +1,395 @@
+---
+modeline: |
+  vim: set ft=pandoc:
+description: |
+    The VirtualBox Packer builder is able to create VirtualBox virtual machines snapshots
+    and export them in the OVF format, starting from an ISO image.
+layout: docs
+page_title: 'VirtualBox Snapshot - Builders'
+sidebar_current: 'docs-builders-virtualbox-vm'
+---
+
+# VirtualBox Builder (from an existing VM)
+
+Type: `virtualbox-vm`
+
+The VirtualBox Packer builder is able to create
+[VirtualBox](https://www.virtualbox.org/) virtual machines snapshots and
+(optionally) export them in the OVF format, starting from an **existing**
+virtual machine.
+
+The builder builds a virtual machine snapshot by using an existing virtual
+machine, booting it, provisioning software within the OS, then shutting it down.
+The result of the VirtualBox builder is a new snapshot persisting all changes
+from the applied provisioners.
+
+## Basic Example
+
+Here is a basic example. which serves to show the basic configuration:
+
+``` json
+{
+  "type"                  : "virtualbox-vm",
+  "communicator"          : "winrm",
+  "headless"              : "{{user `headless`}}",
+  "winrm_username"        : "vagrant",
+  "winrm_password"        : "vagrant",
+  "winrm_timeout"         : "2h",
+  "shutdown_command"      : "shutdown /s /t 10 /f /d p:4:1 /c \"Packer Shutdown\"",
+  "guest_additions_mode"  : "disable",
+  "output_directory"      : "./builds-vm",
+  "vm_name"               : "target-vm",
+  "attach_snapshot"       : "Snapshot",
+  "target_snapshot"       : "Target-Snapshot",
+  "force_delete_snapshot" : "true",
+  "keep_registered"       : "false",
+  "skip_export"           : "false"
+}
+```
+
+It is important to add a `shutdown_command`. By default Packer halts the virtual
+machine and the file system may not be sync'd. Thus, changes made in a
+provisioner might not be saved.
+
+## Configuration Reference
+
+There are many configuration options available for the VirtualBox builder. They
+are organized below into two categories: required and optional. Within each
+category, the available options are alphabetized and described.
+
+In addition to the options listed here, a
+[communicator](/docs/templates/communicator.html) can be configured for this
+builder.
+
+### Required:
+
+-   `vm_name` (string) - This is the name of the virtual machine to which the
+    builder shall attach.
```

Comment:
Based on my testing, target_snapshot is also required, right?

Reply:
Megan, I'm afraid you're right

https://github.com/hashicorp/packer/blob/c0e37e604563f87b22576b8df07147b562a1e3a0/builder/virtualbox/vm/config.go#L100

But the documentation is correct because this was, and still is, my intention to make the `target_snapshot` setting optional. The raeson is that, if you only want to create an export after all provisioners run you don't need a (new) snapshot to persist the changes.

I'll file a new PR for this.

Reply:
@SwampDragons Hi Megan! As promised I provided a patch (PR) for the issue that the `target_snapshot` configuration parameter wasn't optional as described in the documentation.

https://github.com/hashicorp/packer/pull/8011